packages/drafts-realm/garden-design.gts

```diff
@@ -249,26 +258,51 @@ class Isolated extends Component<typeof GardenDesign> {
     let ids: string[] = [];
     for (let row = 0; row < this.rows; row++) {
       for (let col = 0; col < this.cols; col++) {
-        ids.push(`r${row}-c${col}`);
+        ids.push(this.composeId(row, col));
       }
     }
     return ids;
   };
 
+  composeId(x: number, y: number) {
+    return `r${x}-c${y}`;
+  }
+
+  decomposeId(id: string): { x: number; y: number } {
+    let arr = id.replace('r', '').replace('c', '').split('-');
+    if (!arr[0] || !arr[1]) {
+      throw new Error('Unknown id format');
```

Comment:
Would be good to be more exact, i.e 
```throw new Error(`Unknown id format: ${id}. Expected format: rx-cy`);``` 

(I wanted to make this as a code suggestion but for some reason i can't find the suggestion button  🤷‍♂️)